uwrt_mars_rover_neopixel/src/uwrt_mars_rover_neopixel_can.cpp

```diff
@@ -0,0 +1,47 @@
+#include "uwrt_mars_rover_neopixel/uwrt_mars_rover_neopixel_can.h"
+
+constexpr uint16_t NEOPIXEL_CAN_ID_INCOMING = 0x785;
+
+NeopixelCan::NeopixelCan(uint16_t c_i, uint8_t fpl, const char* name) : _addr{}, _ifr{} {
+  // Prepare the outgoing can packet
+  _outgoing_packet.can_id = c_i;
+  _outgoing_packet.can_dlc = fpl;
+  // General work for socket binding
+  _ifname = name;
+  if ((_s = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+    ROS_ERROR("Error while opening socket\n");
+    throw - 1;
+  }
+  strcpy(_ifr.ifr_name, _ifname);
+  ioctl(_s, SIOCGIFINDEX, &_ifr);
+  _addr.can_family = AF_CAN;
+  _addr.can_ifindex = _ifr.ifr_ifindex;
+  // Bind socket CAN to an interface
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast): reinterpret cast required by syscall
+  if (bind(_s, reinterpret_cast<struct sockaddr*>(&_addr), sizeof(_addr)) < 0) {
+    ROS_ERROR("Error in socket bind");
+    throw - 2;
+  }
+}
+void NeopixelCan::sendCAN(const uint8_t data) {
+  // Store data into the data potion of the CAN packet
+  _outgoing_packet.data[0] = data;
+  // Send out the CAN packet
+  write(_s, &_outgoing_packet, sizeof(_outgoing_packet));
+}
+bool NeopixelCan::waitforAck() {
+  ROS_INFO("Now waiting for acknowledgement message.");
+  bool first_pass = true;
```

Comment:
no need for a first pass variable. should just have an if statement at the end that checks if the packet contents is valid